### DISCUSSION TITLE
enh: toggle option for group users to make models public

### DISCUSSION THREAD

**Discussion by tkg61** [open-webui/open-webui#8054] [Ideas]
# Feature Request

---

**Context**
In 0.4.x users are able to make their own models inside of a group. These models can either be made public for all users in openwebui or private for a specific group. 

**Describe the solution you'd like**
A administrative setting/toggle set at each individual group (but with a default option available) to do the following: 

  - Allow users in a group to publish models publicly for all users
  - Prevent users from making their models public and only allow them to set it to a specific group of their choosing

**Describe alternatives you've considered**
None

**Additional context**
Essentially this is creating the ability for more of a private workspace only for a user in a group instead of a pseudo-admin style privilege

**Comment by tkg61** [comment]
i realized this should probably also be extended to prompts, tools, and knowledge as well.

**Comment by spammenotinoz** [comment]
Building on from this, ideally would like Workplace Models to behave like GPTs.
- Allow users to create Models (AKA GPT's), with custom knowledge.
- For privacy reasons, allow anyone to use the Model\GPT but only the creator or admin can edit or view the knowledge article.
